### PR TITLE
Feature/preview in floating window

### PR DESCRIPTION
**Pull request by PhilippOesch**
Because I'm a big fan of the floating window, but also of the preview functionality.

This implements #303

### CODE REVIEW

File: lua/oil/config.lua [hunk 1]
```diff
@@ -121,6 +121,10 @@ local default_config = {
     win_options = {
       winblend = 0,
     },
+    -- gap between oil window and preview window
+    preview_gap = 2,
+    -- preview_split: Split direction: "left", "right", "above", "below".
+    preview_split = 'left',
```
Comment (stevearc) [inline lua/oil/config.lua:127]: I think the default should be `"auto"`, which choose left/right based on the value of `vim.o.splitright`

**Review by stevearc — CHANGES_REQUESTED** [review]
Visually looks great! Needs stylua applied and some of the tests are no longer passing. I also noticed some bugs if you open oil + preview, then `:tabnew` and open oil + preview again.

Will provide a more thorough code review after these things are addressed

**Comment by stevearc** [comment]
@PhilippOesch I moved some of the logic around and did some other various refactoring. Take a look, kick the tires, and if you're happy with it I'll merge!

**Comment by PhilippOesch** [comment]
> @PhilippOesch I moved some of the logic around and did some other various refactoring. Take a look, kick the tires, and if you're happy with it I'll merge!

The refactoring looks good from my view. Thank you for taking your time and looking at it. If you want, you can merge.

**Pull request merged**